crypto/src/key/mod.rs

```diff
@@ -140,6 +163,81 @@ impl From<MLRistrettoPrivateKey> for PrivateKey {
     }
 }
 
+impl ExtendedPrivateKey {
+    pub fn new_master(
+        seed: &[u8],
+        key_kind: ExtendedKeyKind,
+    ) -> Result<ExtendedPrivateKey, DerivationError> {
+        match key_kind {
+            ExtendedKeyKind::Secp256k1Schnorr => {
+                Ok(Secp256k1ExtendedPrivateKey::new_master(seed)?.into())
+            }
+        }
+    }
+
+    pub fn new_from_entropy(key_kind: ExtendedKeyKind) -> (ExtendedPrivateKey, ExtendedPublicKey) {
+        Self::new_from_rng(&mut make_true_rng(), key_kind)
+    }
+
+    pub fn new_from_rng(
+        rng: &mut (impl Rng + CryptoRng),
+        key_kind: ExtendedKeyKind,
+    ) -> (ExtendedPrivateKey, ExtendedPublicKey) {
+        match key_kind {
+            ExtendedKeyKind::Secp256k1Schnorr => {
+                let k = Secp256k1ExtendedPrivateKey::new(rng);
+                (k.0.into(), k.1.into())
+            }
+        }
+    }
+
+    pub fn kind(&self) -> ExtendedKeyKind {
+        match self.key {
+            ExtendedPrivateKeyHolder::Secp256k1Schnorr(_) => ExtendedKeyKind::Secp256k1Schnorr,
+        }
+    }
+
+    pub(crate) fn get_internal_key(&self) -> &ExtendedPrivateKeyHolder {
+        &self.key
+    }
+}
+
+impl ExtendedPublicKey {
+    pub fn kind(&self) -> ExtendedKeyKind {
+        match self.pub_key {
+            ExtendedPublicKeyHolder::Secp256k1Schnorr(_) => ExtendedKeyKind::Secp256k1Schnorr,
+        }
+    }
+
+    pub fn get_internal_key(&self) -> &ExtendedPublicKeyHolder {
+        &self.pub_key
+    }
+
+    pub fn from_private_key(private_key: &ExtendedPrivateKey) -> ExtendedPublicKey {
+        match private_key.get_internal_key() {
+            ExtendedPrivateKeyHolder::Secp256k1Schnorr(ref k) => {
+                Secp256k1ExtendedPublicKey::from_private_key(k).into()
+            }
+        }
+    }
+}
+
+impl From<Secp256k1ExtendedPrivateKey> for ExtendedPrivateKey {
```

Comment:
This trait impl shouldn't exist. The struct `Secp256k1ExtendedPrivateKey` is not publicly accessible, and this is only used in new. So to save 20 more characters in the `new` function(s), we're adding a whole trait impl. Not a good trade-off. 

Same applies to `Secp256k1ExtendedPublicKey`.